llvm/test/tools/llvm-rc/Inputs/split-path.rc

```diff
@@ -0,0 +1,2 @@
+100 ICON "subdir" "/icon-new.ico"
```

Comment:
Would it be better if a preprocessor definition was used here instead of `"subdir"`, and below in the test? Like shown in https://github.com/llvm/llvm-project/issues/51286? That would show the end-intention for someone who reads this patch later.

Reply:
That'd be nicer - but we can't use the preprocessor in these tests. We're building llvm-rc as part of llvm, and llvm tests can't depend on clang. So all the llvm-rc tests run with `-no-preprocessor` to skip that step.

Reply:
Ah yes I see. I wonder if the following would work in another .test file:
```
# REQUIRES: llvm-driver, clang

# RUN: %llvm llvm-rc ...
```
That would only run under
`cmake ... -DLLVM_ENABLE_PROJECTS="llvm;clang" -DLLVM_TOOL_LLVM_DRIVER_BUILD=ON`
but I think it's fine we have other tests like this.

Not sure though if we support launching `clang` from `llvm-rc` like this, but that'd be a nice to have eventually!

Reply:
Oh, interesting concept! Anyway, I don't think there's enough added value to complicate the testing setup here, even if it would show the intended test case clearer. I guess I could add a comment in either the `.test` or `.rc` to explain clearer what scenario the test replicates.

(We already have a few tests of llvm-rc in clang, to showcase how it successfully manages to call the preprocessor there though, but I wouldn't go so far as to place the testcase for this feature on the clang side, for the sake of keeping tests small and localized.)